plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java

```diff
@@ -17,27 +21,45 @@
  */
 @Extension
 @Restricted(NoExternalUse.class)
-public class HudsonPrivateSecurityRealmConfigurator extends DataBoundConfigurator {
+public class HudsonPrivateSecurityRealmConfigurator extends DataBoundConfigurator<HudsonPrivateSecurityRealm> {
+    private static final Logger logger = LoggerFactory.getLogger(HudsonPrivateSecurityRealmConfigurator.class);
+    /// matches HudsonPrivateSecurityRealm.JBCRYPT_HEADER, requires Jenkins >= 164.1
+    private static final String HASHED_PASSWORD_PREFIX = "#jbcrypt:";
 
     public HudsonPrivateSecurityRealmConfigurator() {
         super(HudsonPrivateSecurityRealm.class);
     }
 
     @Override
-    public Set<Attribute> describe() {
-        final Set<Attribute> describe = super.describe();
+    public Set<Attribute<HudsonPrivateSecurityRealm, ?>> describe() {
+        final Set<Attribute<HudsonPrivateSecurityRealm, ?>> describe = super.describe();
         describe.add(new MultivaluedAttribute<HudsonPrivateSecurityRealm, UserWithPassword>("users", UserWithPassword.class)
-            .getter(target ->
-                target.getAllUsers().stream()
-                    .map(u -> new UserWithPassword(u.getId(), null))    // password isn't actually stored, only hashed
-                    .collect(Collectors.toList()))
-            .setter((target, value) -> {
-                for (UserWithPassword user : value) {
+            .getter(HudsonPrivateSecurityRealmConfigurator::getter)
+            .setter(HudsonPrivateSecurityRealmConfigurator::setter)
+        );
+        return describe;
+    }
+
+    private static Collection<UserWithPassword> getter(HudsonPrivateSecurityRealm target) {
```

Comment:
A `Collection` is not the best choice for generic collections, and I'd suggest `List` as that is what you are returning in the `toList()` at the return statement.
Reason being `Collection` is too abstract

```suggestion
    private static List<UserWithPassword> getter(HudsonPrivateSecurityRealm target) {
```

Reply:
The interface for `MultivaluedAttribute` is `Attribute<Owner, Collection<Type>>`. I could extend something different, but I didn't want to rewrite the code that much for this feature.

Reply:
I must have missed that. My bad